src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -367,69 +403,101 @@ export function useSwapInputsController({
     const params = buildQuoteParams({
       currentAddress: store.getState().settings.accountAddress,
       inputAmount,
-      outputAmount,
       inputAsset: internalSelectedInputAsset.value,
+      lastTypedInput: lastTypedInputParam,
+      outputAmount,
       outputAsset: internalSelectedOutputAsset.value,
-      lastTypedInput,
     });
 
     logger.debug(`[useSwapInputsController]: quote params`, {
       data: params,
     });
 
     if (!params) {
-      runOnUI(resetFetchingStatus)(true);
+      runOnUI(resetFetchingStatus)({ fromError: true });
       return;
     }
 
-    const response = (params.swapType === SwapType.crossChain ? await getCrosschainQuote(params) : await getQuote(params)) as
-      | Quote
-      | CrosschainQuote
-      | QuoteError;
-
-    logger.debug(`[useSwapInputsController]: quote response`, {
-      data: response,
-    });
-
-    // TODO: Handle native asset inputs
-    runOnUI(setQuote)({
-      data: response,
-      outputAmount:
-        lastTypedInput === 'inputAmount'
+    try {
+      const [quoteResponse, fetchedPrices] = await Promise.all([
+        params.swapType === SwapType.crossChain ? getCrosschainQuote(params) : getQuote(params),
+        fetchAssetPrices({
+          inputAsset: internalSelectedInputAsset.value,
+          outputAsset: internalSelectedOutputAsset.value,
+        }),
+      ]);
+
+      const quotedInputAmount =
```

Comment:
this parsing of the `quoteResponse` for `quotedInputAmount` and `quotedOutputAmount` could live in `setQuote` since it's receiving the `quoteResponse` as well as these params anyway